hal/ubuntu/qemu.sh

```
@@ -17,264 +21,310 @@ PASSWORD="password"
+    local user_data_file="$1"
+
+    echo "Creating SNP seed image with $user_data_file..."
+
+    if ! command -v genisoimage &> /dev/null; then
+        echo "genisoimage is not installed. Please install genisoimage and try again."
+        exit 1
+    fi
+
+    local cidata_dir
+    cidata_dir="$(mktemp -d)"
+    trap "rm -rf $cidata_dir" RETURN
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)